### ISSUE TITLE
Questions about upgrading cluster-autoscaler from 1.23 to 1.28

### ISSUE THREAD

**Issue by duyawen8** [kubernetes/autoscaler#6972]
**Which component are you using?**: cluster-autoscaler

**Is your feature request designed to solve a problem? If so describe the problem this feature should solve.**:

**Describe the solution you'd like.**:

**Describe any alternative solutions you've considered.**:

**Additional context.**:

I want to upgrade CA from 1.23 to 1.28 to use the [parallel drain](https://github.com/kubernetes/autoscaler/blob/2284649f5aa3ba9813779cf390c49100e9c1a9be/cluster-autoscaler/proposals/parallel_drain.md) feature. The kubernetes version is 1.21. Is there any compatibility issue?

**Comment by Shubham82** [comment]
Hi @duyawen8, there will be a compatibility issue, if you use k8s v1.21 for CA 1.28. For CA 1.28, you have to use k8s v1.28.
See the [release](https://github.com/kubernetes/autoscaler/tree/master/cluster-autoscaler#releases) section under README.md for the corresponding k8s version to the CA version.

**Comment by Shubham82** [comment]
@duyawen8, if your concern is resolved so can we close this issue?

**Comment by duyawen8** [comment]
I know that the CA version and the K8s version have a one-to-one correspondence. Is there any way to use CA 1.28 for parallel eviction without upgrading the cluster? Are there definitely compatibility issues?

**Comment by Shubham82** [comment]
IMO there will be compatibility issues because For every CA release, we update corresponding upstream dependencies(k8s).
so there might be issues with mismatched and deprecated API items, I don't think it's recommended to use a CA that is greater than your cluster version.

**Comment by Shubham82** [comment]
cc @gjtempleton @MaciekPytel
your thoughts on this?

**Comment by Shubham82** [comment]
> IMO there will be compatibility issues because For every CA release, we update corresponding upstream dependencies(k8s). so there might be issues with mismatched and deprecated API items, I don't think it's recommended to use a CA that is greater than your cluster version.

cc @MaciekPytel @gjtempleton @jackfrancis 
WDYT?

**Comment by jackfrancis** [comment]
tl;dr here is that this particular scenario is not tested by the project, so we can't officially say either way, although the general guidance and compatibility matrix would suggest the guidance "not recommended"

if upgrading your cluster is not a practical possibility, you could stress test this in a staging environment to tease out the behaviors, and whether or not any API connectivity pathologies are meet your operational requirements

I think this is about as clear of a message as we can give about this particular request. Not recommended but it's true that there is a possibility that within any given k8s release delta that falls outside of the supported and tested compatibility matrix your cluster may work acceptably for you.

/close
